py/torch_tensorrt/dynamo/conversion/impl/topk.py

```diff
@@ -155,6 +158,37 @@ def topk(
             k,
             get_axes_for_reduce_op(get_positive_dim(dim, len(input.shape))),
         )
+    if k == DYNAMIC_DIM:
+        output_shape = get_shape_with_dynamic_shape(
+            ctx, target, source_ir, name, input.shape, input
+        )
+        layer = ctx.net.add_slice(
+            output_shape,
+            start=[dim],
+            shape=[1],
+            stride=[1],
+        )
+        set_layer_name(layer, target, name)
+
+        # Get scalar tensor from 1d tensor
+        shuffle_layer = ctx.net.add_shuffle(layer.get_output(0))
+        shuffle_layer.reshape_dims = trt.Dims()
+        set_layer_name(shuffle_layer, target, name, source_ir)
+
+        cond = le(
+            ctx,
+            target,
+            source_ir,
+            f"{name}_k_cond",
+            shuffle_layer.get_output(0),
+            TRT_TOPK_MAX_ELEMENT,
+        )
+        ctx.net.add_assertion(
```

Comment:
when dynamic k > 3840, there is segmentation fault rather than runtime error. Error is like below

Python error: Segmentation fault

Thread 0x00007fda56bfe640 (most recent call first):
  File "/root/.pyenv/versions/3.10.14/lib/python3.10/threading.py", line 324 in wait
  File "/root/.pyenv/versions/3.10.14/lib/python3.10/threading.py", line 607 in wait
  File "/root/.pyenv/versions/3.10.14/lib/python3.10/site-packages/tqdm/_monitor.py", line 60 in run
  File "/root/.pyenv/versions/3.10.14/lib/python3.10/threading.py", line 1016 in _bootstrap_inner
  File "/root/.pyenv/versions/3.10.14/lib/python3.10/threading.py", line 973 in _bootstrap

Current thread 0x00007fdc02f65740 (most recent call first):
  File "/root/trt/TensorRT/py/torch_tensorrt/dynamo/runtime/_PythonTorchTensorRTModule.py", line 287 in forward
...
Extension modules: numpy.core._multiarray_umath, ...
Segmentation fault

I tried catch it from add_assertion() but it doesn't work while simple test with add_assertion() was ok.
Any idea to use it? Or we can with this behavior to enable dynamic k value for sort?
There is no problem if dynamic k value is in range as in test code

Reply:
I think the issue is due to [TRT's limitation](https://docs.nvidia.com/deeplearning/tensorrt/archives/tensorrt-1000-ea/api/python_api/infer/Graph/Layers.html#tensorrt.ITopKLayer). One possible solution is to fall back to pytorch if k > 3840. We don't expect to throw any errors due to TRT

Reply:
Thanks for review. As we discuss, I changed to allow this ops when k value is static because we cannot identify dynamic k value at compile time.